src/main/kotlin/uk/bfi/uvaudit/event/AuditEventController.kt

```diff
@@ -7,6 +7,8 @@ import org.springframework.http.converter.HttpMessageNotReadableException
 import org.springframework.security.core.annotation.AuthenticationPrincipal
 import org.springframework.web.bind.annotation.*
 import uk.bfi.uvaudit.security.AuditUser
+import javax.naming.AuthenticationException
```

Comment:
I think you want [`org.springframework.security.core.AuthenticationException`](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/AuthenticationException.htmll) here, `javax.naming.AuthenticationException`refers to authentication errors with JNDI or directory services.

Reply:
`org.springframework.security.core.AuthenticationException` is abstract so I added a custom exception type `EmailNotVerifiedException` which inherits from it